From e2c7b4d962f88baaf13029de96b69667045dd436 Mon Sep 17 00:00:00 2001
From: Joel Yliluoma <joel.yliluoma@iki.fi>
Date: Sun, 28 Jan 2018 15:08:29 +0200
Subject: [PATCH 2/2] Expose cycle-limit interface through the TR2 CPU register

---
 src/cpu/cpu.cpp | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/cpu/cpu.cpp b/src/cpu/cpu.cpp
index 363b091..633e1a6 100644
--- a/src/cpu/cpu.cpp
+++ b/src/cpu/cpu.cpp
@@ -1695,6 +1695,14 @@ bool CPU_READ_DRX(Bitu dr,Bit32u & retvalue) {
 }
 
 bool CPU_WRITE_TRX(Bitu tr,Bitu value) {
+	if(tr == 2) /* Bisqwit hack: Write into TR2 changes cycle limit */
+	{
+	    CPU_CycleMax        = value;
+	    CPU_OldCycleMax     = value;
+	    CPU_CycleAutoAdjust = false;
+	    GFX_SetTitle(CPU_CycleMax,-1,false);
+	    return false;
+	}
 	/* Check if privileged to access control registers */
 	if (cpu.pmode && (cpu.cpl>0)) return CPU_PrepareException(EXCEPTION_GP,0);
 	switch (tr) {
@@ -1711,6 +1719,11 @@ bool CPU_WRITE_TRX(Bitu tr,Bitu value) {
 }
 
 bool CPU_READ_TRX(Bitu tr,Bit32u & retvalue) {
+        if(tr == 2) /* Bisqwit hack: Put cycle limiter in TR2 */
+        {
+                retvalue = CPU_CycleMax;
+                return false;
+        }
 	/* Check if privileged to access control registers */
 	if (cpu.pmode && (cpu.cpl>0)) return CPU_PrepareException(EXCEPTION_GP,0);
 	switch (tr) {
-- 
2.15.1

